src/elements/element.rectangle.js

```diff
@@ -50,11 +50,106 @@ module.exports = function(Chart) {
 	}
 
 	Chart.elements.Rectangle = Chart.Element.extend({
-		draw: function() {
+		getBorderWidth: function(top, right, bottom, left) {
+			// Get calculated border width
+			var vm = this._view;
+			// borderWidth should be less than bar width and bar height.
+			// ..,maximum border width for left-right borders
+			var barSizeLeftRight = Math.abs(left - right);
+			// ...maximum border width for top-bottom borders
+			var barSizeTopBottom = Math.abs(top - bottom);
+
+			function ceiling(val, max) {
+				return val > max ? max : val;
+			}
+
+			// Like CSS:
+			// - one border to rule them all
+			// - top-bottom + left-right
+			// - top + right + bottom + left
+			return !vm.borderWidth ? [] :
+				vm.borderWidth instanceof Array && vm.borderWidth.length === 1 ? {
```

Comment:
This whole block would be more compact and efficient if `borderWidth` is an object.